rep-2002.rst

```diff
@@ -0,0 +1,144 @@
+The release manager should then setup the target rosdistro with all the appropriate configurations, such as target platforms and architectures.
+The tool will then iterate through all repositories in the snapshot of the source rosdistro and attempt to automatically run bloom release on the package at the specific version in the snapshot.
+If it fails the maintainer will be needed to resolve any issues.
+
+If the process gets interrupted the migration tool can be rerun with the same source snapshot and target rosdistro.
+The migration tool will skip any package already in the target rosdistro.
+The rerun is valuable in cases when a maintainer has resolved an an issue and many more downstream repositories are unblocked, as well as if the operation gets interrupted.
+
+After the snapshot is taken maintainers must make releases for both the source rosdistro and target rosdistro as develoment is considered forked.
+
+
+Releasing stable ROS distributions from a rolling release
+---------------------------------------------------------
+
+Prior to formally starting development for the next stable release, the rolling release will be used as a staging ground for new content.
+At an announced time prior to start the official stable release, a snapshot of the rolling release will be created and will be used to boostrap the next stable release development process.
+This will be automated by the migration tool outlined above.
+
+
+Updating the rolling release platforms
+--------------------------------------
+
+The rolling release is intended to remain available indefinitely and so will not be retired or reach end-of-support status without an update to this REP and the ROS distribution release process.
+However, this does mean that platform changes during the life of the rolling distribution are going to occur and the rolling release may target platforms that are themselves unstable or in a pre-release state.
+When a platform change is coming, the intended date of change will be announced ahead of time.
+
+The platform change is a unique invocation of the migration tool where the source and target rosdistro will be the same.
+To do this the person managing the migration will take the snapshot of the source rosdistro and then clear it's contents as well as updating the target platforms.
+After which the migration tool can be invoked with the target and source rosdistro as the same, because the source rosdistro will be leveraging the snapshot.
+
+
+Bootstrapping the rolling release
+---------------------------------
+
+The initial creation of the rolling rosdistro will use the rolling release tools "in reverse" to create the rolling release from the May 2020 ROS 2 Foxy Fitzroy release.
+Package maintainers may then continue to release into either or both Foxy and the rolling release distribution independently.
+
+
+Recommendations for package maintainers
+=======================================
+
+Versioning between rolling and stable distributions
```

Comment:
I appreciate having recommendations, but the current text feels a bit confusing, if not incomplete. Probably because I'm trying to read it from the semver point of view.

Would perhaps a more general recommendation start with:

* When making backwards compatible releases, it's recommended to target both the stable and rolling releases and use the same version number for both.
* When making changes that should go only into the rolling release, consider bumping the minor or major version numbers, not just the patch number.

Reply:
I tried to synthesize this feedback with the specific case I want to recommend be avoided (interleaved or overlapping version numbers) in 2b285c5 . What do you think?

Reply:
Much clearer, thanks!